error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFluxOperation.java

```diff
@@ -0,0 +1,130 @@
+            + "please make sure that they are used with this in mind.",
+    summary = "Accidental blocking of `Flux` with convenience method.",
+    link = BUG_PATTERNS_BASE_URL + "ImplicitBlockingFluxOperation",
+    linkType = CUSTOM,
+    severity = ERROR,
+    tags = LIKELY_ERROR)
+public class ImplicitBlockingFluxOperation extends BugChecker
+    implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<ExpressionTree> FLUX_TO_ITERABLE =
+      instanceMethod()
+          .onDescendantOf(Suppliers.typeFromString("reactor.core.publisher.Flux"))
+          .named("toIterable")
+          .withNoParameters();
+  private static final Matcher<ExpressionTree> FLUX_TO_STREAM =
+      instanceMethod()
+          .onDescendantOf(Suppliers.typeFromString("reactor.core.publisher.Flux"))
+          .named("toStream")
+          .withNoParameters();
+  private static final Matcher<ExpressionTree> FLUX_IMPLICIT_BLOCKING_METHOD =
+      anyOf(FLUX_TO_ITERABLE, FLUX_TO_STREAM);
+
+  /** Instantiates a new {@link ImplicitBlockingFluxOperation} instance. */
+  public ImplicitBlockingFluxOperation() {}
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (!FLUX_IMPLICIT_BLOCKING_METHOD.matches(tree, state)) {
+      return Description.NO_MATCH;
+    }
+
+    Description.Builder description = buildDescription(tree);
+
+    description.addFix(getSuppressWarningsFix(state));
+    if (ThirdPartyLibrary.GUAVA.isIntroductionAllowed(state)) {
+      description.addFix(getGuavaFix(tree, state));
+    }
+    description.addFix(getUnmodifiableListFix(tree, state));
+
+    return description.build();
+  }
+
+  private static SuggestedFix getSuppressWarningsFix(VisitorState state) {
+    return SuggestedFixes.addSuppressWarnings(state, "ImplicitBlockingFluxOperation");
+  }
+
+  private static SuggestedFix getGuavaFix(MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder guavaFix = SuggestedFix.builder();
+    String toImmutableList =
+        SuggestedFixes.qualifyStaticImport(
+            "com.google.common.collect.ImmutableList.toImmutableList", guavaFix, state);
+    return getCollectAndBlockFix(tree, state, guavaFix, toImmutableList + "()").build();
+  }
+
+  private static SuggestedFix getUnmodifiableListFix(
+      MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder unmodifiableListFix = SuggestedFix.builder();
+    String toUnmodifiableList =
+        SuggestedFixes.qualifyStaticImport(
+            "java.util.stream.Collectors.toUnmodifiableList", unmodifiableListFix, state);
+    return getCollectAndBlockFix(tree, state, unmodifiableListFix, toUnmodifiableList + "()")
+        .build();
+  }
+
+  /**
+   * merges `flux.collect(...).block()...` fix into given fix with specified collector and postfix
+   * to match the original expression tree.
+   *
+   * @param collector expression
```

Comment:
Note to self: this Javadoc is not really corresponding with the actual parameters 😉. Also, we add a `.` after every sentence.